src/lib/Underground.js

```diff
@@ -438,4 +455,43 @@ class AutomationUnderground
 
         return itemsState;
     }
+
+    /**
+     * @brief Check for daily deals opportunities to make more diamonds and trade those that are beneficial
+     */
+    static __internal__tradeDiamonds()
+    {
+        const deals = DailyDeal.list()
+        let dealsDone = 0;
+        let totalProfit = 0;
+        for (const [ i, deal ] of deals.entries())
+        {
+            // Only consider deals for items that both have a Diamond value
+            if ((deal.item1.valueType == UndergroundItemValueType.Diamond)
```

Comment:
We should check is the item is not locked as well here

Reply:
Well, diamonds for diamonds is almost always right, unless you are waiting for an even better trade
But plates for diamonds MIGHT not be wanted, as the player might prefer to get the gems, and fossils might also not be wanted to trade.

The destination item must be a diamond item or the trades probably do not make much sense.
If we allow for source non-diamond items, the value should be patched to 0, as the "value" for the plates is set to 100 (gems)

I did not remember that locking items in the underground list was a thing, and I agree that a check there is needed. Only for the source item, right? And what to do if the item is a fossil? Those cannot be locked, and according to the code, are valid items for a trade.

If you still want me to do both changes, let me know and I will push them to my branch

Reply:
About the first point, as a first intention it sounds good as-is. We might consider to add more settings to enable more trading to happen (including automatically sell items for example).

---

About the second point, I'd prefer to have the locking implemented before merging, since it might have unwanted side-effects.

It should indeed only be done for the source item. Lets not bother with items that cannot be locked.

You can push the modification, but be sure to pull my changes before you do such modification, to avoid loosing the improvements I made.

If you don't know how to do it, here are the commands :
```shell
# Make a backup of your branch for safety
git checkout -b save auto-mine-trade

# Force the checkout of the server's modifications (⚠️ this will drop any changes you made without pushing)
git fetch && git checkout -B auto-mine-trade origin/auto-mine-trade 
```

Reply:
I already have both changes prepared, so I will push both.

A minor inconvenience is that you cannot lock an item that you have 0 units of, so once the automation sells all of them, you have to disable the selling to be able to lock the item again.

But as both selling options are almost always right, and they can be easily disabled, I dont think this is a problem

Reply:
Yeah, that's why I prefer not to implement this right now

Reply:
> If you don't know how to do it, here are the commands :

Thanks, I did not know that!

VSCode did manage it for me, though, so I did not need to use them.